eng/pipelines/templates/jobs/ci.yml

```diff
@@ -226,18 +228,51 @@ jobs:
         parameters:
           BaselineFilePath: $(Build.sourcesdirectory)/eng/dotnet.gdnbaselines
           ServiceDirectory: ${{ parameters.ServiceDirectory }}
-
   - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
     parameters:
+      GenerateJobName: generate_target_service_test_matrix
       JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
       MatrixConfigs: ${{ parameters.MatrixConfigs }}
       MatrixFilters: ${{ parameters.MatrixFilters }}
       MatrixReplace: ${{ parameters.MatrixReplace }}
       CloudConfig:
-        Cloud: Public
+        Cloud: public
       AdditionalParameters:
         SDKType: ${{ parameters.SDKType }}
-        ServiceToTest: ${{ coalesce(parameters.ServiceToTest, parameters.ServiceDirectory) }}
         ServiceDirectory: ${{ parameters.ServiceDirectory }}
         TestSetupSteps: ${{ parameters.TestSetupSteps }}
-        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }} 
+  - ${{ if ne(parameters.TestDependsOnDependency, 'not-specified') }}:
+    - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+      parameters:
+        GenerateJobName: generate_target_dependencies_test_matrix 
+        JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+        MatrixConfigs: 
+          - Name: NET_ci_test_base
+            # The path here should refer to dependency.tests.yml parameter ProjectFilesOutputFolder 
+            Path: $(Build.ArtifactStagingDirectory)/platform-matrix.json
```

Comment:
Should this be `${{ parameters.MatrixFilters }}`? Right now this seems to be forcing it to always use the default matrix.

Reply:
I am currently not overwriting the original one but generate a new platform json. 
Otherwise, every time we run them in local, the original `platform-matrix.json` will change and very likely to checkin to the changes. Also, it is risky to mutual the file which will abundant finally.

Reply:
I understand you are creating a new file but some pipelines pass in a custom matrix so we should be starting with that matrix instead of hardcoding the default matrix here.